enhancements/alert-on-operator-updates.md

```diff
@@ -0,0 +1,158 @@
+services) running across a Kubernetes cluster. Given that the installed Operators have the freedom to do almost anything with appropriate access privileges, there is a 
+potential threat that these actions may cause harm to the cluster, are malicious, or detrimental to a cluster upgrade. 
+
+This enhancement deals with handling two such Operator actions:  
+- Alert is triggered when the operator deploys operands with invalid PDB([Pod Disruption Budget](https://kubernetes.io/docs/tasks/run-application/configure-pdb/#specifying-a-poddisruptionbudget)) that would cause an openshift upgrade issue
+- Alert is triggered when an operator tries to modify an existing on-board SCC([Security Context Constraints](https://www.openshift.com/blog/understanding-service-accounts-sccs))
+ 
+
+### Goals
+
+1. Provide a mechanism to alert when an invalid PDB is being applied on to a cluster 
+2. Prevent operators from modifying existing SCCs in the cluster
+3. Provide a configuration option to cluster admins for a permissive mode vs. a enforcing mode
+    - Alert from the start and add enforcement after we get data on how many of these alerts are firing.
+    - Able to update independently what is an alerting rule and what is an enforcing rule. 
+    - Provide option to turn off an enforcing mode
+
+## Open Questions 
+
+1. Configuration option on OLM to make for a permissive mode vs. a enforcing mode 
+   
+2. Validation library is used to define what components of an Operator are deemed potentially harmful 
+
+
+## Design Details
+
+The design is split into 3 sections, each one per goal defined above. 
+
+## 1. A mechanism to alert when an invalid PDB is being applied on to a cluster (a.k.a validating PDB yaml) 
+
+Let's consider an example PDB.yaml
+
+```yaml
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
+metadata:
+  name: zk-pdb
+spec:
+  minAvailable: 2  
+  selector:
+    matchLabels:
+      app: zookeeper
+```
+
+PodDisruptionBudget has three fields:
+
+- A label selector `.spec.selector` to specify the set of pods to which it applies. This field is required.
+- `.spec.minAvailable` which specifies the number of pods that must still be available after the eviction, even in the absence of the evicted pod.
+- `.spec.maxUnavailable` which specifies the number of pods that can be unavailable after the eviction
+
+Note: `minAvailable` & `maxUnavailable` can be either an absolute number or a percentage.
+
+An PDB is considered invalid if any of these fields has incorrect values and gets applied on to a cluster.
+
+Below lists a number of potential approaches:
+
+### 1. Add a command line flag (Straw Man proposal)
+The flag parses the PDB yaml and validates the fields 
+ 
+### 2. Validate PDB same way as the yaml files under `/manifests` folder in validation library (Straw Man proposal)
+This is similar to how OLM's existing manifests are validated as part of the validation library
```

Comment:
I would say this method aligns with what’s done already. We validate all the manifests in the bundle.

Reply:
So, could we possibly follow the same? If not, what is the argument here?

Reply:
Sorry, this is confusing now. Are we talking about manifests shipped with the operator? Or are we actually talking about objects generated by the operator *at runtime*. Those seem like two totally separate problems. At the moment, OLM doesn't allow arbitrary kube manifests in the bundle itself, and unless that happens then OLM wouldn't be vaildating yaml at all.

Reply:
I clarified this with Shawn. You are right that we are talking about operator actions and not the configuration of the yaml. I'll update this